cleanup.js

```diff
@@ -48,18 +60,75 @@ function _validateParameters() {
   }
 }
 
-//Entrypoint of the cleanup script
-msRestAzure.loginWithServicePrincipalSecret(clientId, secret, domain, function (err, credentials) {
-  if (err) return console.log(err);
-  resourceClient = new ResourceManagementClient(credentials, subscriptionId);
-  computeClient = new ComputeManagementClient(credentials, subscriptionId);
-  deleteVirtualMachine(function (err, result) {
-    if (err) return console.log('Error occured in deleting the virtual machine: ' + vmName + '\n' + util.inspect(err, { depth: null }));
-    console.log('Successfully deleted the virtual machine: ' + vmName);
-    console.log('\nDeleting the resource group can take few minutes, so please be patient :).');
-    deleteResourceGroup(function (err, result) {
-      if (err) return console.log('Error occured in deleting the resource group: ' + resourceGroupName + '\n' + util.inspect(err, { depth: null }));
-      console.log('Successfully deleted the resourcegroup: ' + resourceGroupName);
+function fetchEndpointMetadata() {
+  // Setting URL and headers for request
+  var options = {
+    url: fetchUrl,
+    headers: {
+      'User-Agent': 'request'
+    },
+    rejectUnauthorized: false
+  };
+  // Return new promise 
+  return new Promise(function (resolve, reject) {
+    // Do async job
+    request.get(options, function (err, resp, body) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(JSON.parse(body));
+      }
+    })
+  })
+}
+
+function main() {
+  var endpointData = fetchEndpointMetadata();
+  endpointData.then(function (result) {
+    var metadata = result;
+    console.log("Initialized user details");
+    // Use user details from here
+    console.log(metadata)
+    map["name"] = "AzureStack"
+    map["portalUrl"] = metadata.portalEndpoint 
+    map["resourceManagerEndpointUrl"] = armEndpoint 
+    map["galleryEndpointUrl"] = metadata.galleryEndpoint 
+    map["activeDirectoryEndpointUrl"] = metadata.authentication.loginEndpoint.slice(0, metadata.authentication.loginEndpoint.lastIndexOf("/") + 1) 
+    map["activeDirectoryResourceId"] = metadata.authentication.audiences[0] 
+    map["activeDirectoryGraphResourceId"] = metadata.graphEndpoint 
+    map["storageEndpointSuffix"] = "." + armEndpoint.substring(armEndpoint.indexOf('.'))  
```

Comment:
storageEndpointSuffix and KeyvaultEndpointSuffix can be read directly from metadata endpoint without parsing from armendpoint. The api version needed for metadata endpoint call  is 2019-10-01 @TheOnlyWei